example/src/tests/groupTests.ts

```diff
@@ -957,6 +958,52 @@ test('can remove and add members from a group by inbox id', async () => {
   return true
 })
 
+test('can cancel streams', async () => {
+  const [alix, bo] = await createClients(2)
+  let messageCallbacks = 0
+
+  await bo.conversations.streamAllMessages(async () => {
+    messageCallbacks++
+  }, true)
+
+  const group = await alix.conversations.newGroup([bo.address])
+  await group.send('hello')
+  await delayToPropogate()
+
+  assert(
+    messageCallbacks === 1,
+    'message stream should have received 1 message'
+  )
+
+  await bo.conversations.cancelStreamAllMessages()
+  await delayToPropogate()
+
+  await group.send('hello')
+  await group.send('hello')
+  await group.send('hello')
+
+  await delayToPropogate()
+
+  assert(
+    messageCallbacks === 1,
+    'message stream should still only received 1 message'
+  )
+
+  await bo.conversations.streamAllMessages(async () => {
+    messageCallbacks++
+  }, true)
+
+  await group.send('hello')
+  await delayToPropogate()
+
+  assert(
+    messageCallbacks === 2,
+    'message stream should have received 2 message'
+  )
+
+  return true
+})
+
```

Comment:
@nplasterer  was this test failing before? I thought streamAllMessages WAS canceled even if underlying stream was not and so this test would have passed before?

Reply:
Yes the test passed before as well. But this shows the logs easier which is how I was debugging.